OSDOCS-10991 OCP 4.12.60 Release Notes (#77965)
openshift/openshift-docs: release_notes/ocp-4-12-release-notes.adoc

[id="ocp-4-12-60"]
=== RHSA-2024:4006 - {product-title} 4.12.60 bug fix and security update

Issued: 2024-06-27

{product-title} release 4.12.60, which includes security updates, is now available. The list of bug fixes that are included in the update is documented in the link:https://access.redhat.com/errata/RHSA-2024:4006[RHSA-2024:4006] advisory. The RPM packages that are included in the update are provided by the link:https://access.redhat.com/errata/RHSA-2024:4008[RHSA-2024:4008] advisory.

You can view the container images in this release by running the following command:

[source,terminal]
----
$ oc adm release info 4.12.60 --pullspecs
----

[id="ocp-4-12-60-bug-fixes"]
==== Bug fixes

* Previously, for the `cluster:capacity_cpu_cores:sum` metric, nodes with the`infra` role but not `master` role were not assigned a value of `infra` for the `label_node_role_kubernetes_io` label. With this update, nodes with the `infra` role but not `master` role are now correctly labeled as `infra` for this metric. (link:https://issues.redhat.com/browse/OCPBUGS-35558[*OCPBUGS-35558*])

> * Previously, when an IngressController was configured with client SSL/TLS, but did not have the `clientca-configmap` finalizer, the Ingress Operator would try to add the finalizer without checking whether the IngressController was marked for deletion. Consequently, if an IngressController was configured with client SSL/TLS and was subsequently deleted, the Operator would correctly remove the finalizer. It would then repeatedly, and erroneously, try and fail to update the IngressController to add the finalizer back, resulting in error messages in the Operator's logs.

Reviewer: ```suggestion
* Previously, when an Ingress Controller was configured with client SSL/TLS, but did not have the `clientca-configmap` finalizer, the Ingress Operator would try to add the finalizer without checking whether the Ingress Controller was marked for deletion. Consequently, if an Ingress Controller was configured with client SSL/TLS and was subsequently deleted, the Operator would correctly remove the finalizer. It would then repeatedly try and fail to update the IngressController to add the finalizer back, resulting in error messages in the Operator's logs.
```
See the repo glossary entry on the [Ingress Controller](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/term_glossary.adoc#ingress-controller). Also, I suggest cutting _erroneously_ as it is fluff.

+

> With this update, the Ingress Operator no longer adds the `clientca-configmap` finalizer to an IngressController that is marked for deletion. As a result, the Ingress Operator no longer tries to perform erroneous updates, and no longer logs the associated errors. (link:https://issues.redhat.com/browse/OCPBUGS-35027[*OCPBUGS-35027*])

Reviewer: ```suggestion
With this update, the Ingress Operator does not add the `clientca-configmap` finalizer to an Ingress Controller that is marked for deletion. As a result, the Ingress Operator does not attempt incorrect updates and does not log the associated errors. (link:https://issues.redhat.com/browse/OCPBUGS-35027[*OCPBUGS-35027*])
```

> * Previously, timeout values larger than what Golang could parse were not properly validated. Consequently, timeout values larger than what HAProxy could parse caused issues with HAProxy. With this update, if the timeout specifies a values larger than what can be parsed, it is capped at the maximum that HAProxy can parse. As a result, issues are no longer caused for HAProxy. (link:https://issues.redhat.com/browse/OCPBUGS-33432[*OCPBUGS-33432*])

Reviewer: ```suggestion
* Previously, timeout values larger than what Golang could parse were not properly validated. Consequently, timeout values larger than what HAProxy could parse caused issues with HAProxy. With this update, if the timeout specifies a value larger than what can be parsed, the value is capped at the maximum value that HAProxy can parse. As a result, issues are no longer caused by HAProxy. (link:https://issues.redhat.com/browse/OCPBUGS-33432[*OCPBUGS-33432*])
```
